### PR TITLE
2172 Update InjectorIDQ

### PR DESCRIPTION
**Pull request by GerardThevenon**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo [hunk 1]
```diff
@@ -17,22 +17,27 @@ model PLL "Phase-Locked Loop"
   import Dynawo.Types;
   import Dynawo.Electrical.SystemBase;
 
+  //Parameters
   parameter Types.PerUnit Kp "PLL proportional gain";
   parameter Types.PerUnit Ki "PLL integrator gain";
   parameter Types.PerUnit OmegaMinPu "Lower frequency limit in pu (base OmegaNom)";
   parameter Types.PerUnit OmegaMaxPu "Upper frequency limit in pu (base OmegaNom)";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo:24]: Alphabetical order : Ki, Kp, OmegaMaxPu, OmegaMinPu

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo [hunk 2]
```diff
@@ -28,11 +28,13 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
 
   parameter Types.ApparentPowerModule SNom "Nominal apparent power in MVA";
 
-  // Inputs: d-q axis pu variables (base UNom, SNom)
+  // Inputs: d-q axis pu variables (base UNom, SNom) and rotor angle
   Modelica.Blocks.Interfaces.RealInput idPu(start = Id0Pu) "Injected d-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, 61}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealInput iqPu(start = Iq0Pu) "Injected q-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, -41}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UPhase(start = UPhase0) "Rotor angle: angle between machine rotor frame and port phasor frame" annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo:36]: precise unit : in rad

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo [hunk 3]
```diff
@@ -16,21 +16,21 @@ model HvdcVSCDanglingP "HVDC VSC model with terminal2 connected to a switched-of
   extends AdditionalIcons.Hvdc;
 
   import Modelica;
-  import Dynawo.Electrical.Sources;
+  import Dynawo;
   import Dynawo.Electrical.HVDC;
   import Dynawo.Types;
-  import Dynawo.Connectors;
   import Dynawo.Electrical.SystemBase;
   import Dynawo.Electrical.Constants;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo:23]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo [hunk 4]
```diff
@@ -16,21 +16,21 @@ model HvdcVSCDanglingP "HVDC VSC model with terminal2 connected to a switched-of
   extends AdditionalIcons.Hvdc;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo:16]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo [hunk 5]
```diff
@@ -16,21 +16,21 @@ model HvdcVSCDanglingP "HVDC VSC model with terminal2 connected to a switched-of
...
+  Dynawo.Connectors.ACPower terminal1(V(re(start = u10Pu.re), im(start = u10Pu.im)), i(re(start = i10Pu.re), im(start = i10Pu.im))) "Connector used to connect the injector to the grid" annotation(
+    Placement(visible = true, transformation(origin = {-150, -8}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Connectors.ACPower terminal2 "Connector used to connect the injector to the grid (switched off side)" annotation(
+    Placement(visible = true, transformation(origin = {150, -8}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
 
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsBaseActivePowerControl;
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsBlockingFunction;
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsACVoltageControl;
+  extends Dynawo.Electrical.Controls.WECC.Parameters.ParamsPLL;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingP.mo:33]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo [hunk 6]
```diff
@@ -16,22 +16,22 @@ model HvdcVSCDanglingUdc "HVDC VSC model with terminal2 connected to a switched-
   extends AdditionalIcons.Hvdc;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo:16]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo [hunk 7]
```diff
@@ -16,22 +16,22 @@ model HvdcVSCDanglingUdc "HVDC VSC model with terminal2 connected to a switched-
...
 
   import Modelica;
-  import Dynawo.Electrical.Sources;
+  import Dynawo;
   import Dynawo.Electrical.HVDC;
   import Dynawo.Types;
-  import Dynawo.Connectors;
   import Dynawo.Electrical.SystemBase;
   import Dynawo.Electrical.Constants;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo:23]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo [hunk 8]
```diff
@@ -16,22 +16,22 @@ model HvdcVSCDanglingUdc "HVDC VSC model with terminal2 connected to a switched-
...
+  Dynawo.Connectors.ACPower terminal1(V(re(start = u10Pu.re), im(start = u10Pu.im)), i(re(start = i10Pu.re), im(start = i10Pu.im))) "Connector used to connect the injector to the grid" annotation(
+    Placement(visible = true, transformation(origin = {-150, -8}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Connectors.ACPower terminal2 "Connector used to connect the injector to the grid (switched off side)" annotation(
+    Placement(visible = true, transformation(origin = {150, -8}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
 
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsBaseDCVoltageControl;
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsBlockingFunction;
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsACVoltageControl;
   extends HVDC.HvdcVSC.BaseControls.Parameters.ParamsDCLine;
+  extends Dynawo.Electrical.Controls.WECC.Parameters.ParamsPLL;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSCDanglingUdc.mo:34]: same as in HVDCVSC.mo

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo [hunk 9]
```diff
@@ -16,68 +16,67 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
...
-    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 5}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
   Modelica.Blocks.Interfaces.RealOutput PInjPuSn(start = -P0Pu * SystemBase.SnRef /SNom) "Injected active power in pu (base SNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 43}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealOutput QInjPu(start = -Q0Pu) "Injected reactive power in pu (base SnRef)";
-  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
+  Modelica.Blocks.Interfaces.RealOutput QInjPuSn(start = -Q0Pu * SystemBase.SnRef /SNom) "Injected reactive power in pu (base SNom)" annotation(
+    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 5}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo:45]: Thanks for putting these outputs in the alphabetical order. Moreover, could you :

- Add the convention  for all four outputs : "Injected active power in pu (base SnRef) (generator convention)"
- Insert a space in '/SNom', which would become '/ SNom'

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo [hunk 10]
```diff
@@ -16,68 +16,67 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
...
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, -79}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
 
-  parameter Types.ApparentPowerModule SNom "Nominal apparent power in MVA";
-
-  // Inputs: d-q axis pu variables (base UNom, SNom)
+  // Inputs: d-q axis pu variables (base UNom, SNom) and rotor angle
   Modelica.Blocks.Interfaces.RealInput idPu(start = Id0Pu) "Injected d-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, 61}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealInput iqPu(start = Iq0Pu) "Injected q-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, -41}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo:35]: (base SNom, UNom) (generator convention)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo [hunk 11]
```diff
@@ -16,68 +16,67 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
...
-
-  parameter Types.VoltageModulePu U0Pu "Start value of voltage amplitude at injector terminal in pu (base UNom)";
-  parameter Types.Angle UPhase0 "Start value of voltage angle at injector terminal in rad";
-  parameter Types.ActivePowerPu P0Pu "Start value of active power in pu (base SnRef) (receptor convention)";
-  parameter Types.ReactivePowerPu Q0Pu "Start value of reactive power in pu (base SnRef) (receptor convention)";
-  parameter Types.ComplexVoltagePu u0Pu "Start value of complex voltage at injector terminal in pu (base UNom)";
-  parameter Types.ComplexApparentPowerPu s0Pu "Start value of apparent power at injector terminal in pu (base SnRef) (receptor convention)";
   parameter Types.ComplexCurrentPu i0Pu "Start value of complex current at injector terminal in pu (base UNom, SnRef) (receptor convention)";
   parameter Types.CurrentModulePu Id0Pu "Start value of idPu in pu (base SNom, UNom)";
   parameter Types.CurrentModulePu Iq0Pu "Start value of iqPu in pu (base SNom, UNom)";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo:53]: (base SNom, UNom) (generator convention)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ_INIT.mo [hunk 12]
```diff
@@ -17,25 +17,26 @@ model InjectorIDQ_INIT "Initialisation model for the injector controlled by d an
...
   parameter Types.ActivePowerPu P0Pu "Start value of active power in pu (base SnRef) (receptor convention)";
   parameter Types.ReactivePowerPu Q0Pu "Start value of reactive power in pu (base SnRef) (receptor convention)";
+  parameter Types.VoltageModulePu U0Pu "Start value of voltage amplitude at injector terminal in pu (base UNom)";
+  parameter Types.Angle UPhase0 "Start value of voltage angle at injector terminal (in rad)";
 
-  Types.ComplexVoltagePu u0Pu "Start value of complex voltage at injector terminal in pu (base UNom)";
-  Types.ComplexApparentPowerPu s0Pu "Start value of complex apparent power at injector terminal in pu (base SnRef) (receptor convention)";
   flow Types.ComplexCurrentPu i0Pu "Start value of complex current at injector terminal in pu (base UNom, SnRef) (receptor convention)";
   Types.PerUnit Id0Pu "Start value of id in pu (base SNom)";
   Types.PerUnit Iq0Pu "Start value of iq in pu (base SNom)";
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ_INIT.mo:27]: (base SNom, UNom) (generator convention)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo [hunk 13]
```diff
@@ -17,22 +17,23 @@ model PLL "Phase-Locked Loop"
   import Dynawo.Types;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo:17]: General comments for this model. 
add.y is basically upu.im*cosphi - uPu.re*sinphi so you can replace all of this with a real expression.
Maybe gain + limInt could be replaced with a limited PI (to be checked)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Photovoltaics/WECC/PVCurrentSource.mo [hunk 14]
```diff
@@ -16,27 +16,28 @@ model PVCurrentSource "WECC PV model with a current source as interface with the
   import Modelica;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Photovoltaics/WECC/PVCurrentSource.mo:16]: Remember we only keep one PLL (the new one)

File: examples/DynaSwing/WECC/Wind/WECCWTG4ACurrentSource/reference/outputs/curves/curves.csv [hunk 15]
```diff
@@ -1,4 +1,4 @@
-time;WTG4A_WTG4A_injector_PInjPu;WTG4A_WTG4A_injector_QInjPu;WTG4A_WTG4A_injector_terminal_V_re;WTG4A_WTG4A_injector_terminal_i_re;WTG4A_WTG4A_injector_terminal_V_im;WTG4A_WTG4A_injector_terminal_i_im;WTG4A_WTG4A_pll_omegaPLLPu;WTG4A_WTG4A_driveTrainPmConstrant_PePu;
+time;WTG4A_WTG4A_injector_PInjPu;WTG4A_WTG4A_injector_QInjPu;WTG4A_WTG4A_injector_terminal_V_re;WTG4A_WTG4A_injector_terminal_i_re;WTG4A_WTG4A_injector_terminal_V_im;WTG4A_WTG4A_injector_terminal_i_im;WTG4A_WTG4A_pll1_omegaPLLPu;WTG4A_WTG4A_driveTrainPmConstrant_PePu;
```
Comment (cossartque) [inline examples/DynaSwing/WECC/Wind/WECCWTG4ACurrentSource/reference/outputs/curves/curves.csv:1]: Please update the references too, thanks

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo (hunk 11 above)
Comment (GerardThevenon) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo:53]: Is it really a current module?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo (hunk 13 above)
Comment (GerardThevenon) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/PLL/PLL.mo:17]: LimitedPI is not equivalent (the location of the limiter is different)

**Comment by cossartque** [comment]
Build fails

**Comment by cossartque** [comment]
Can you squash once everything is ready ?

File: dynawo/sources/Models/Modelica/Dynawo/Examples/HVDC/HVDC.mo [hunk 16]
```diff
@@ -41,10 +41,10 @@ model HVDC "HVDC link connected to two infinite buses"
   Modelica.Blocks.Sources.Step QRef2Pu(height = 0, offset = 0.301, startTime = 1) annotation(
     Placement(visible = true, transformation(origin = {50, 80}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
   Modelica.Blocks.Sources.Step URef2Pu(height = 0, offset = 1.079767, startTime = 4.5) annotation(
-    Placement(visible = true, transformation(origin = {50, 49}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
+    Placement(visible = true, transformation(origin = {50, 50}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
   Modelica.Blocks.Sources.Step UdcRefPu(height = 0.0, offset = 1, startTime = 6) annotation(
     Placement(visible = true, transformation(origin = {50, 20}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
-  Dynawo.Electrical.Events.NodeFault nodeFault(RPu = 0.00, XPu = 0.0075, tBegin = 0.5, tEnd = 1.5) annotation(
+  Dynawo.Electrical.Events.NodeFault nodeFault(RPu = 0.00, XPu = 0.075, tBegin = 0.5, tEnd = 1.5) annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Examples/HVDC/HVDC.mo:47]: Should we keep these values?

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSC.mo [hunk 17]
```diff
@@ -70,44 +70,49 @@ model HvdcVSC "HVDC VSC model"
...
   parameter Types.VoltageModulePu Udc20Pu "Start value of dc voltage at terminal 2 in pu (base UdcNom)";
-  parameter Types.PerUnit Ip20Pu "Start value of active current at terminal 2 in pu (base SNom)";
-  parameter Types.PerUnit Iq20Pu "Start value of reactive current at terminal 2 in pu (base SNom)";
-  parameter Real modeU20 "Start value of the real assessing the mode of the control at terminal 2: 1 if U mode, 0 if Q mode";
   parameter Types.VoltageModulePu UdcRef0Pu "Start value of dc voltage reference in pu (base UdcNom)";
+  parameter Types.Angle UPhase10 "Start value of voltage angle at terminal 1 (in rad)";
+  parameter Types.Angle UPhase20 "Start value of voltage angle at terminal 2 (in rad)";
 
 equation
+  Theta1 = Modelica.Math.atan2(terminal1.V.im, terminal1.V.re);
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcVSC/HvdcVSC.mo:110]: You can put : Theta1 = phi1 and Theta2 = phi2. And make Theta1 and Theta2 outputs of the model HvdcVSC.
Verify the nrt.

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

**Review by cossartque — APPROVED** [review]

**Pull request merged**